### PR TITLE
Limit the number of connections to server

### PR DESCRIPTION
**Pull request by jtlisi**
We ran into a situation where it would be good to limit the total number of connections. This PR adds an optional configuration to limit the number of simultaneous connections to the server. The package used to enable this is the [netutil LimitListener](https://godoc.org/golang.org/x/net/netutil#LimitListener).

Signed-off-by: Jacob Lisi <jacob.t.lisi@gmail.com>

### CODE REVIEW

**Comment by tomwilkie** [comment]
@bboreham mind taking a look?  LGTM.

File: Gopkg.toml [hunk 1]
```diff
@@ -25,3 +25,7 @@
 [[constraint]]
   name = "github.com/gogo/status"
   version = ">=v1.0.3"
+
+[[constraint]]
```
Comment (bboreham) [inline Gopkg.toml]: Please add a comment explaining the need for this constraint, and the conditions under which it can be removed.

File: Gopkg.lock [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "aws/endpoints",
     "internal/ini",
     "internal/sdkio",
-    "internal/shareddefaults",
+    "internal/shareddefaults"
```
Comment (bboreham) [inline Gopkg.lock]: This seems like a spurious and unnecessary change.  Do you know where it came from?

File: server/server.go [hunk 3]
```diff
@@ -60,8 +63,10 @@ type Config struct {
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.HTTPListenAddress, "server.http-listen-address", "", "HTTP server listen address.")
 	f.IntVar(&cfg.HTTPListenPort, "server.http-listen-port", 80, "HTTP server listen port.")
+	f.IntVar(&cfg.HTTPConnLimit, "server.http-conn-limit", 0, "Maximum number of simultaneously http connections, <=0 to disable")
```
Comment (bboreham) [inline server/server.go]: s/simultaneously/simultaneous/

File: Gopkg.lock (hunk 2 above)
Comment (jtlisi) [reply Gopkg.lock]: This was done automatically up running `dep ensure`

File: server/server.go (hunk 3 above)
Comment (jtlisi) [reply server/server.go]: 👍

**Comment by jtlisi** [comment]
The constraint was added automatically by dep for some reason. However, it does not seem to be required so I removed it and re-ran `dep ensure`.

@bboreham This should be good for a second look
